### PR TITLE
More energy related changes

### PR DESCRIPTION
**Pull request by ArnoutD**
Mostly the one hour gap is now solved on address rotation 
Lowered the Node timeout to 5 seconds from 20....
Can run up to PARALLEL_LEVEL 8

We might have to discuss how/when to handle Stick/Node Exceptions

### CODE REVIEW

**Comment by bouwew** [comment]
@ArnoutD there are conficts, I think you need to rebase your PR onto the latest version of the async-branch.

File: plugwise_usb/nodes/__init__.py [hunk 1]
```diff
@@ -536,57 +554,39 @@ def _node_info_update_state(
...
                 )
-            )
+            except (StickError, NodeError) as e:
+                _LOGGER.warning(
+                    "%s for ping_update() for %s",
+                    str(e),
+                    self.mac
+                )
+                await self._available_update_state(False)
+                return False
```
Comment (brefra) [inline plugwise_usb/nodes/__init__.py:591]: The ping_update function (just like any other *_update function) should return the typed result. So in this case the populated `NetworkStatistics` dataclass object, of None if it failed.

File: plugwise_usb/nodes/__init__.py [hunk 2]
```diff
@@ -536,57 +554,39 @@ def _node_info_update_state(
...
         if ping_response is None:
             await self._available_update_state(False)
-            return None
+            return False
```
Comment (brefra) [inline plugwise_usb/nodes/__init__.py:594]: The ping_update function (just like any other *_update function) should return the typed result. So in this case the populated `NetworkStatistics` dataclass object, of None if it failed.

File: plugwise_usb/nodes/__init__.py [hunk 3]
```diff
@@ -423,16 +428,29 @@ async def node_info_update(
     ) -> bool:
```
Comment (brefra) [inline plugwise_usb/nodes/__init__.py:441]: To be consistent the node_info_update function (just like the other *_update functions) should return the populated `NodeInfo `dataclass object or None if it failed.

File: plugwise_usb/nodes/circle.py [hunk 4]
```diff
@@ -297,11 +298,15 @@ async def energy_update(
...
-            _LOGGER.debug(
+            _LOGGER.info(
                 "async_energy_update | Log rollover for %s",
                 self._node_info.mac,
             )
             if await self.node_info_update():
+                # if the log_address changes big chance we did not see slot 4, request it once
+                if self._previous_log_address:
+                    await self.energy_log_update(self._previous_log_address)
+                    self._previous_log_address = None
```
Comment (brefra) [inline plugwise_usb/nodes/circle.py]: I discovered yesterday the log address at the `NodeInfoRespone` message is actually the pointer to the current log period which is not populated with the counters yet.
See also [this commit in the async branch](https://github.com/plugwise/python-plugwise-usb/commit/f6cc26d1104f072dfef298360470f26712fe1763)

File: plugwise_usb/nodes/circle.py [hunk 5]
```diff
@@ -834,7 +840,11 @@ async def node_info_update(
                 self.mac
             )
         if self._last_log_address != node_info.last_logaddress:
+            self._previous_log_address = self._last_log_address
+            _LOGGER.debug('%s update last_log_address %d -> %d', self.mac, 
+                          self._last_log_address, node_info.last_logaddress)            
             self._last_log_address = node_info.last_logaddress
```
Comment (brefra) [inline plugwise_usb/nodes/circle.py]: I discovered yesterday the log address at the `NodeInfoRespone` message is actually the pointer to the current log period which is not populated with the counters yet.
See also [this commit in the async branch](https://github.com/plugwise/python-plugwise-usb/commit/f6cc26d1104f072dfef298360470f26712fe1763)

File: plugwise_usb/nodes/helpers/pulses.py
Comment (brefra) [inline plugwise_usb/nodes/helpers/pulses.py:1]: I did a rewrite on the rollover part in the async branch already.
it was not working correctly and had some bugs for edge cases like when no logging is available or a rollover of the log address itself. So I do not think these changes are required anymore.
So in the rebase I skipped these changes.

File: plugwise_usb/connection/receiver.py [hunk 6]
```diff
@@ -127,72 +129,63 @@ def connection_made(self, transport: SerialTransport) -> None:
...
         self._transport.close()
+        self._msg_processing_task.cancel()
 
     def data_received(self, data: bytes) -> None:
         """Receive data from USB-Stick connection.
 
         This function is called by inherited asyncio.Protocol class
         """
-        _LOGGER.debug("USB stick received [%s]", data)
+        #_LOGGER.info("USB stick received [%s]", data)
```
Comment (brefra) [inline plugwise_usb/connection/receiver.py:152]: Uncomment code.
Lets log it either as debug or remove not at all.

**Review by brefra — CHANGES_REQUESTED** [review]
Sorry, I added some (conflicting) commits to the async branch before.
I tried to do a rebase/merge as best as I could. I did drop most of the changes in the pulses.py file, as those where related to the power log rollover. 
For which I fixed some real bugs already in the async branch:-

-  the log address at the `NodeInfoRespone` message is actually the pointer to the current log period which is not yet populated with the counters yet.
- The range of the log address pointer is 0 to 6015, so it takes that in account too.
- Take empty log records in account to

So I do not think most of these changes at pulses.py are required anymore.
Left some other small comments in the code

File: plugwise_usb/nodes/helpers/pulses.py
Comment (ArnoutD) [reply plugwise_usb/nodes/helpers/pulses.py:1]: yes was rebasing on that but will look what to keep

File: plugwise_usb/connection/receiver.py (hunk 6 above)
Comment (ArnoutD) [reply plugwise_usb/connection/receiver.py:152]: yes it's more debug^2 on the data layer, will remove them ... unless we want something to record dataflow for testing

**Review by brefra — COMMENTED** [review]
There are still some pylint errors due to merge conflicts issue which blocks the merge.

Because there are still some relevant commits to the communication part I've cherry-picked those into [cherry-pick-async_fixes](https://github.com/plugwise/python-plugwise-usb/tree/cherry-pick-async_fixes) branch.
Lets drop this PR and go ahead at the other branch

**Pull request closed without merging**